src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -159,6 +186,10 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 		return this._onLanguageFlavorChanged.event;
 	}
 
+	public get providerNameToDisplayNameMap(): { [providerDisplayName: string]: string } {
```

Comment:
When you return it, is it possible to mark it as readonly, like
```typescript
{ readonly [providerDisplayName: string]: string };
```
so that consumers cannot mess up your internal collection?